app-modules/authorization/src/Http/Controllers/SocialiteController.php

```diff
@@ -46,9 +49,18 @@ public function callback(SocialiteProvider $provider)
             return redirect()->to(Filament::getLoginUrl());
         }
 
+        if ($provider === SocialiteProvider::Azure) {
+            $request = Http::withToken($socialiteUser->token)
+                ->contentType('image/jpeg')
+                ->get('https://graph.microsoft.com/v1.0/me/photo/$value');
+
+            $user->addMediaFromString($request->body())->usingFileName(Str::uuid() . '.jpg')->toMediaCollection('avatar');
+        } else {
+            $user->addMediaFromUrl($socialiteUser->getAvatar())->toMediaCollection('avatar');
+        }
```

Comment:
After thinking about this a little more, I think if we DO have the image as a URL we should use the URL rather than download the image to save on storage space in our application.

We could change this here into a match statement and if it Azure then save to the media collection, if it is Google then we can save the avatar URL in the users table column.

Then add something like a "getAvatar" method on the User model that returns either the URL saved in the DB or, if that is null, a generated URL for the media item.

That way we are taking advantage of external hosting of the asset when we can, saving some resource cost on our end.

Reply:
`default` of the match could just return `null` to prevent errors just in case.

Reply:
I reintroduced the url behavior. There's no reason to have the match with just the one option here because I doubt we will have other SSO providers that do not return a url. Just checking the property is a trivial operation and trying to split the update will just result in more boilerplate.